src/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -680,47 +703,54 @@ std::unique_ptr<model_spec> activity_classifier::init_model() const
       /* stride_height */ 1,
       /* stride_width */ prediction_window,
       /* padding */ padding_type::VALID,
-      /* weight_init_fn */
-      xavier_weight_initializer(num_features * prediction_window,
-                                NUM_CONV_FILTERS * prediction_window,
-                                &random_engine),
+      /* weight_init_fn */ initializer,
       /* bias_init_fn */ zero_weight_initializer());
   result->add_relu("relu1", "conv");
 
   result->add_channel_slice("hiddenIn","stateIn",0,LSTM_HIDDEN_SIZE,1);
   result->add_channel_slice("cellIn","stateIn",LSTM_HIDDEN_SIZE,LSTM_HIDDEN_SIZE*2,1);
+
+  if (use_random_init) {
+    lstm_initializer = lstm_weight_initializers::create_with_xavier_method(
+        NUM_CONV_FILTERS, LSTM_HIDDEN_SIZE, &random_engine);
+  }
   result->add_lstm(
-      /* name */                "lstm",
-      /* input */               "relu1",
-      /* hidden_input */        "hiddenIn",
-      /* cell_input */          "cellIn",
-      /* hidden_output */       "hiddenOut",
-      /* cell_output */         "cellOut",
-      /* input_vector_size */   NUM_CONV_FILTERS,
-      /* output_vector_size */  LSTM_HIDDEN_SIZE,
+      /* name */ "lstm",
+      /* input */ "relu1",
+      /* hidden_input */ "hiddenIn",
+      /* cell_input */ "cellIn",
+      /* hidden_output */ "hiddenOut",
+      /* cell_output */ "cellOut",
+      /* input_vector_size */ NUM_CONV_FILTERS,
+      /* output_vector_size */ LSTM_HIDDEN_SIZE,
       /* cell_clip_threshold */ LSTM_CELL_CLIP_THRESHOLD,
-      /* initializers */  lstm_weight_initializers::create_with_xavier_method(
-          NUM_CONV_FILTERS, LSTM_HIDDEN_SIZE, &random_engine));
+      /* initializers */ lstm_initializer);
```

Comment:
If running clang-format on this existing code, could we add spaces inside the comments so that the string constants continue to line up? The main goal here is to keep the code readable

Reply:
hmmm it is not smart enough to do this.
Will add spaces by my self.